src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -130,9 +130,6 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
         private async ValueTask SynchronizePrimaryWorkspaceAsync(CancellationToken cancellationToken)
         {
             var solution = _workspace.CurrentSolution;
-            if (solution.BranchId != _workspace.PrimaryBranchId)
-                return;
```

Comment:
totally unnecessary.  we are always accessing _workspace.CurrentSolution above.  So this solution was always the primary-branch solution.